packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -37,6 +40,14 @@ interface IButtonClassNames {
 
 export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState> implements IButton {
 
+  private get _splitButton(): boolean {
+    return (this.props.menuProps != null && this.props.onClick != null);
+  }
+
+  private get _expanded(): boolean {
```

Comment:
It just felt more explicit this way. Using directly `this.state.menuProps !== null;` is not as clear.
Do you still feel like this isn't a good idea @jair-cazarin?